### DISCUSSION TITLE
Could not spawn; error code 8 with canitlever beam example (Windows 11)

### DISCUSSION THREAD

**Discussion by FacyMcGitFace** [snl-dakota/dakota#92] [Q&A] [answered]
Dear Community,

I am currently exploring the use of Dakota and navigating through the screen casts, training materials, and exercises. I have successfully installed both the Dakota GUI and CLI, and when I enter "dakota -version" in the command window, it shows: “Dakota version 6.19 released Nov. 15 2023”. The first Rosenbrook example worked well for me( as shown in this video [Introduction to Dakota: Running a Simple Example in Dakota (youtube.com)](https://www.youtube.com/watch?v=ofi13UTq_Is&list=PLouetuxaIMDo-NMFXT-hlHYhOkePLrayY&index=1&t=190s)):
dakota -input rosen_multidim.in -output rosen_multidim_out.dat

However, I am facing challenges with the  cantilever beam example (as seen in the training video “model characterization” at the 46 minute mark).
The issue arises when I input the following command:
dakota -input dakota_cantilever_centered
 After some Dakota-specific entries, I encounter an error message:

`---------------------
Begin Evaluation    1
---------------------
Parameters for evaluation 1:
                      1.0000000000e+00 w
                      1.0000000000e+00 t
                      1.0000000000e+01 L
                      5.0000000000e+02 p
                      2.9000000000e+07 E
                      5.0000000000e+00 X
                      1.0000000000e+01 Y

blocking fork: driver.sh C:\Users\P6C2C1.VOI\AppData\Local\Temp\dakota_params_50e752cc C:\Users\P6C2C~1.VOI\AppData\Local\Temp\dakota_results_f7b4428d

Could not spawn; error code 8 (Exec format error)`

Unfortunately, I am stuck at this point. What could be the problem? A Google search did not provide a solution.
I’ve read somewhere that changing "fork" to "system" in the +.in file might help. However, when I try this, I receive the following message:
 
`---------------------
Begin Evaluation    1
---------------------
Parameters for evaluation 1:
                      1.0000000000e+00 w
                      1.0000000000e+00 t
                      1.0000000000e+01 L
                      5.0000000000e+02 p
                      2.9000000000e+07 E
                      5.0000000000e+00 X
                      1.0000000000e+01 Y

driver.sh C:\Users\P6C2C1.VOI\AppData\Local\Temp\dakota_params_e6e49f20 C:\Users\P6C2C~1.VOI\AppData\Local\Temp\dakota_results_66d19e08

Error: cannot open results file "C:\Users\P6C2C~1.VOI\AppData\Local\Temp\dakota_results_66d19e08" for evaluation 1`

and the driver.sh file opens automatically in an editor.
As you may have noticed, I have no idea what I’m doing. So I hope someone can assist me. If more information is needed, I am happy to provide it.

Best regards,

Update:

So I just found out that I have to use Windows PowerShell instead of cmd. And to make it work on windows, I have to replace the "fork" with "system" in the input file.
Nevertheless, the mentioned error above still remains. 
I currently working with this example:
https://github.com/snl-dakota/dakota-examples/tree/master/official/centered_parameter_study

The powershell (run as admin) output is this:

`---------------------
Begin Evaluation    1
---------------------
Parameters for evaluation 1:
                      1.0000000000e+00 w
                      1.0000000000e+00 t
                      1.0000000000e+01 L
                      5.0000000000e+02 p
                      2.9000000000e+07 E
                      5.0000000000e+00 X
                      5.0000000000e+00 Y

driver.sh params.in results.out

Error: cannot open results file "workdir\run.1\results.out" for evaluation 1`

**Comment by FacyMcGitFace** [comment, marked as answer]
Hi there,

I just managed to solve the problem myself.
I have to use the examples provided here:
https://github.com/snl-dakota/dakota-examples/tree/master

There is a folder which contains updated examples for the training (e.g. using python3), including hints of what needs to be changed in order to perform the exercises in windows.
Everything works now.

Cheers

**Comment by xgtan02** [comment]
On Windows, Dakota may not be able to execute a Linux C-shell script (e.g. driver.sh). This could result in the error message "Error: cannot open results file results.out.1 for evaluation 1" because driver.sh failed to run and results.out.1 was never created. A simple fix for this issue is to create a batch file driver.bat containing "bash driver.sh %1 %2" and use "analysis_driver = 'driver.bat'" in the Dakota input file.
